client/src/api/index.ts

```diff
@@ -125,3 +133,46 @@ export function hasDetails(entry: DatasetEntry): entry is DatasetDetails {
  * Contains dataset metadata information.
  */
 export type MetadataFiles = components["schemas"]["MetadataFile"][];
+
+export function isHistorySummary(history: AnyHistory): history is HistorySummary {
+    return !("user_id" in history);
+}
+
+export function isHistorySummaryExtended(history: AnyHistory): history is HistorySummaryExtended {
+    return "contents_active" in history && "user_id" in history;
+}
```

Comment:
How do you feel about adding a constant field in the model that indicates what view we're dealing with ?
The pro is that we don't need helper to figure out the actual type, the con is that we're serializing more than strictly necessary.

Reply:
That would simplify things yes, I like the idea, but I didn't consider it because it was changing the API response again :laughing: but I can give it a try and see how we feel about it. Should I target this in a follow-up on `dev` for that change though? Or directly in this PR?

Reply:
Let's target dev and mention it at the backend meeting ? We do have a related precedent in `model_class` which is only used for a similar purpose. Yet another alternative is that we include the model-type in the response header, which doesn't alter the response in the body but that would at least be more explicit than `"contents_active" in history && "user_id" in history`, which feels like a contract we could easily break accidentally.